subprojects/docs/src/docs/userguide/img/software_architecture.puml

```diff
@@ -0,0 +1,52 @@
+@startuml
+
+skinparam monochrome true
+skinparam component {
+    BackgroundColor white
+    BorderColor Black
+	ArrowColor Black
+}
+
+database "External Repository" {
+  [Spring Boot\nFramework] as SpringBoot
+  [Android\nFramework] as Android
+}
+
+package "Application Layer" {
+  [server-application]
+  [android-app]
+}
+
+package "Business Logic Layer" {
+  [user-feature]
+  [admin-feature]
+}
+
+package "Domain Model Layer" {
+  [domain-model] <-right- [state]
+}
+
+[server-application] -down-> [user-feature]
+[server-application] -down-> [admin-feature]
+[android-app] -down-> [user-feature]
+
+[user-feature] -down-> [domain-model]
+[user-feature] -down-> [state]
+
+[admin-feature] -down-> [domain-model]
+[admin-feature] -down-> [state]
+
+[android-app] -down-> [SpringBoot]
+[server-application] -down-> [Android]
```

Comment:
Is this correct? Shouldn't  `android-app` depend on `Android` framework and `server-application` depend on `SpringBoot` framework instead?

Reply:
Yes! Good catch!